src/consensus.rs

```diff
@@ -93,37 +94,50 @@ impl<T: Proposition> Consensus<T> {
     // membership: gen = pending_gen
     /// Handles a signed vote
     /// Returns the vote we cast and the reached consensus vote in case consensus was reached
-    pub fn handle_signed_vote(
-        &mut self,
-        signed_vote: SignedVote<T>,
-        gen: Generation,
-    ) -> Result<VoteResponse<T>> {
+    pub fn handle_signed_vote(&mut self, signed_vote: SignedVote<T>) -> Result<VoteResponse<T>> {
```

Comment:
We no longer pass in `gen: Generation` when calling `handle_signed_vote`, since we assume that the vote's generation is already checked prior to this call, we can simply read the generation from the `signed_vote.vote.gen`.

(this came out of a bug I had introduced while fixing the prop test, I was passing in the wrong generation number which was causing us to produce merge votes with different generation numbers)